development/components/database/prestashopcollection.md

```diff
@@ -0,0 +1,240 @@
+
+```php
+$productCollection->join('categories', 'id_category', PrestaShopCollection::LEFT_JOIN);
+$productCollection->join('categories', 'id_category', PrestaShopCollection::INNER_JOIN);
+$productCollection->join('categories', 'id_category', PrestaShopCollection::LEFT_OUTER_JOIN);
+```
+
+### Using where with join
+
+One of the interest of joining other ObjectModel entities to your collection, is the possibility to filter on the external entity with `where()`. 
+
+```php
+$productCollection = (new PrestaShopCollection('Product'))
+    ->join('manufacturer')
+    ->where('manufacturer.name', '=', 'Manufacturer AAA');
+```
+
+## Ordering a Collection (order by)
+
+To order your collection, use the `orderBy()` method. 
+
+```php
+public function orderBy($field, $order = 'asc')
+```
+
+Ordering can be done in ascending or descending direction, with `asc` or `desc` `$order` parameter.
+
+```php
+$productCollection = (new PrestaShopCollection('Product'))
+    ->orderBy('reference', 'desc');
+```
+
+## Grouping (group by)
+
+To group your collection items on a field, use the `groupBy()` method. 
+
+```php
+public function groupBy($field)
+```
+
+```php
+$productCollection = (new PrestaShopCollection('Product'))
+    ->groupBy('id_supplier');
+```
+
+## Paginating a Collection (offset / limit)
+
+When fetching large quantities of items, you may need to paginate your collection, with SQL `OFFSET / LIMIT` equivalent mecanismes. 
+
+To do so, use those two methods:
+
+```php
+public function setPageNumber($page_number)
+public function setPageSize($page_size)
+```
+
+To use this feature, first, set the $page_size:
+
+```php
+$productCollection = (new PrestaShopCollection('Product'))
+    ->setPageSize(100); // will get only 100 items
+```
+
+If only `$page_size` is set, the collection will return the first `$page_size` items. 
+
+To do a pagination, and get the page 2 of the collection, use the following example: 
+
+```php
+$productCollection = (new PrestaShopCollection('Product'))
+    ->setPageSize(100) // will get only 100 items
+    ->setPageNumber(2); // but from page 2, equivalent to offset=(pageNumber - 1) * page_size. 
+```
+
+- Getting the page: `1` of a collection with a page_size of: `100` is the `PrestaShopCollection` equivalent of this `SQL` statement: 
+    `LIMIT 100` or `LIMIT 100 OFFSET 0` or `LIMIT 0,100`
+
+- Getting the page: `2` of a collection with a page_size of: `100` is the `PrestaShopCollection` equivalent of this `SQL` statement: 
+    `LIMIT 100 OFFSET 100` or `LIMIT 100,100`
+
+- Getting the page: `3` of a collection with a page_size of: `100` is the `PrestaShopCollection` equivalent of this `SQL` statement: 
+    `LIMIT 100 OFFSET 200` or `LIMIT 200,100`
+
+{{% notice info %}}
+Known issue: When not using `$id_lang`, and using pagination (`setPageSize()`, `setPageNumber()`), pagination will be broken. Please use a Lang Context if the entity is multi-lang enabled. 
```

Comment:
I don't understand this part now. I think you were more clear when we discussed it. Pagination is broken when getting multi-lang entries with `$id_lang` provided. Yes? Because you wrote:
`Please use a Lang Context if the entity is multi-lang enabled. `